src/lib/incoming-channel.js

```diff
@@ -28,176 +28,121 @@ module.exports = class IncomingChannel {
     return yield this._balance.get()
   }
 
-  * create ({ hash }) {
+  * create ({ channelId }) {
     // fetches details from the network to initialize
-    this._hash = hash
-    this._tx = yield this._api.connection.request({
-      command: 'tx',
-      transaction: this._hash
-    })
+    this._channelId = channelId
+    yield this._store.put('channel_i', channelId)
 
-    if (this._tx.Destination !== this._address) {
-      throw new Error('channel destination is ' + this._tx.Destination
-        + ' but our address is ' + this._address)
+    const paychan = yield this._api.getPaymentChannel(channelId)
+    if (paychan.destination !== this._address) {
+      throw new Error('channel destination is ' + paychan.destination +
+        ' but our address is ' + this._address)
     }
 
-    this._publicKey = Buffer.from(this._tx.PublicKey, 'hex').slice(1)
-    this._channelId = util.channelId(
-      this._tx.Account,
-      this._tx.Destination,
-      this._tx.Sequence)
-
-    this._maximum = (new BigNumber(this._tx.Amount)).div('1000000')
+    this._publicKey = Buffer.from(paychan.publicKey, 'hex').slice(1)
+    const maximum = new BigNumber(paychan.amount).mul(1000000)
```

Comment:
1 XRP is equal to 1,000,000 XRP drops, which are the base unit of the XRP ledger. I'll make that a constant when I refactor this code.